UnitTests/SqlPreparation/SqlPreparationExpressionVisitorTest.cs

```diff
@@ -715,7 +716,25 @@ public void VisitPartialEvaluationExceptionExpression_StripsExceptionAndVisitsIn
     }
 
     [Test]
-    public void VisitConstantExpression_WithNoCollection_LeavesExpressionUnchanged ()
+    public void VisitConstantExpression_WithNonCollectionEnumerable_ThrowsNotSupportedException ()
+    {
+      var constantExpression = Expression.Constant (Mock.Of<IEnumerable>());
+
+      Assert.That (
+          () =>
+          {
+            SqlPreparationExpressionVisitor.TranslateExpression (
+                constantExpression,
+                _context,
+                _stageMock.Object,
+                _methodCallTransformerProvider);
+          },
+          Throws.InstanceOf<NotSupportedException>().With.Message.EqualTo (
+              "Expressions must not contain unevaluated IEnumerable objects. Use ICollection, ICollection<T>, or IReadOnlyCollection<T> instead."));
```

Comment:
```suggestion
              "Only collection can be used as a parameter. Use an object that implements the ICollection, ICollection<T>, or IReadOnlyCollection<T> interface instead of a sequence based on IEnumerable."));
```